prometheus-to-sd/main.go

```diff
@@ -153,41 +126,13 @@ func readAndPushDataToStackdriver(stackdriverService *v3.Service, gceConf *confi
 			glog.V(4).Infof("Skipping %v component as there are no metric to expose.", sourceConfig.Component)
 			continue
 		}
-
 		metrics, err := translator.GetPrometheusMetrics(sourceConfig.Host, sourceConfig.Port)
-		commonConfig := &config.CommonConfig{
-			GceConfig:     gceConf,
-			PodConfig:     podConfig,
-			ComponentName: sourceConfig.Component,
-		}
 		if err != nil {
 			glog.Warningf("Error while getting Prometheus metrics %v", err)
 			continue
 		}
-		if metricDescriptors != nil {
-			updateMetricDescriptorsDescription(stackdriverService, commonConfig, metricDescriptors, metrics)
-		}
-		ts := translator.TranslatePrometheusToStackdriver(commonConfig, metrics, sourceConfig.Whitelisted)
+		metricDescriptorCache.UpdateMetricDescriptors(metrics, sourceConfig.Whitelisted)
```

Comment:
We want to update this only for custom metrics. I know there is a check underneath but I think it's a bit confusing that you always call this function.